cgp/utils.py

```diff
@@ -108,8 +108,13 @@ def compute_key_from_numpy_evaluation_and_args(
             rng.uniform(_min_value, _max_value, size=_batch_size)
             for _ in range(ind.genome._n_inputs)
         ]
-        y = f_single(*x)
-        s = np.array_str(np.array(y), precision=15)
+        y = np.array(f_single(*x))
+        if not np.any(np.isfinite(y)):
+            raise RuntimeError(
```

Comment:
I wonder if raising an error is a good solution here. For a user this could cause problems, if during a run suddenly for one individual no finite value can be computed the whole run is terminated. 
Could we just skip calculating a key for such an individual? So that if this happens the individual will just be "ignored" for the decorator and evaluated?

Reply:
hmm, i'm not sure we should solve this for the user by default. wouldn't it be better to encourage the user to completely avoid such situations and rather adapt, for example the `min_value, max_value` ranges?

if they explicitly want to protect themselves against such crashes (sounds like a bad idea, but nevertheless) they can use a try except, wrapping the `compute_key...` function:
```python
def my_protected_compute_key(*args, **kwargs):
    try:
        return compute_key_from_numpy_evaluation_and_args(*args, **kwargs)
    except RuntimeError:
        <return a unique identifier which makes sure the cache is not used>
```

this way, the "good practice" is easy (adapt the ranges) while the "bad practice" (continue even though some keys are meaningless) is slightly more difficult.

Reply:
Maybe I am misunderstanding this, but this could happen in (almost) any run, I think? The user has no way of a priori knowing what `f_single` is, so it is hard (impossible?) to define a good range for `x`. 
Let me try to make an example: Take a `genome` with default params (` n_inputs=1, n_outputs=1, n_columns=128, n_rows: int = 1, primitives=Primitives((Add, Sub, Mul, ConstantFloat))`. Now, `f_single(x)=2^128` is a possible function encoded by this genome. If my thinking is correct, this would lead to a RuntimeError (numeric overflow), which is undesirable from a user perspective imo

Reply:
i've thought more about it and i now think you're right. :+1: let's go with a warning here. it would be super annoying if you're 23h42min run is canceled bc of one key that wasn't computed well.